pkg/destroy/openstack/openstack_deprovision.go

```diff
@@ -329,51 +329,59 @@ func deleteRouters(opts *clientconfig.ClientOpts, filter Filter, logger logrus.F
 		os.Exit(1)
 	}
 	for _, router := range allRouters {
-		// Get non HA router interface ports
-		portListOpts := ports.ListOpts{
-			DeviceID:    router.ID,
-			DeviceOwner: "network:router_interface",
+		// If a user provisioned floating ip was used, it needs to be dissociated
+		// Any floating Ip's associated with routers that are going to be deleted will be dissociated
+		fipOpts := floatingips.ListOpts{
+			RouterID: router.ID,
 		}
-		allPagesPort, err := ports.List(conn, portListOpts).AllPages()
+
+		fipPages, err := floatingips.List(conn, fipOpts).AllPages()
 		if err != nil {
 			logger.Fatalf("%v", err)
 			os.Exit(1)
 		}
-		allPorts, err := ports.ExtractPorts(allPagesPort)
+
+		allFIPs, err := floatingips.ExtractFloatingIPs(fipPages)
 		if err != nil {
 			logger.Fatalf("%v", err)
 			os.Exit(1)
 		}
 
-		// Get HA router interface ports
-		HAportListOpts := ports.ListOpts{
-			DeviceID:    router.ID,
-			DeviceOwner: "network:ha_router_replicated_interface",
+		for _, fip := range allFIPs {
+			floatingips.Update(conn, fip.ID, floatingips.UpdateOpts{})
+		}
+
+		// Get router interface ports
+		portListOpts := ports.ListOpts{
+			DeviceID: router.ID,
 		}
-		HAallPagesPort, err := ports.List(conn, HAportListOpts).AllPages()
+		allPagesPort, err := ports.List(conn, portListOpts).AllPages()
 		if err != nil {
 			logger.Fatalf("%v", err)
 			os.Exit(1)
 		}
-		HAPorts, err := ports.ExtractPorts(HAallPagesPort)
+		allPorts, err := ports.ExtractPorts(allPagesPort)
 		if err != nil {
 			logger.Fatalf("%v", err)
 			os.Exit(1)
 		}
 
-		// Catch all, since router may not be HA
-		allPorts = append(allPorts, HAPorts...)
-
+		// map to keep track of whethere interface for subnet was already removed
+		subnets := make(map[string]bool)
```

Comment:
nit: the variable name could be a little more descriptive, for instance removedSubnets